docs/data/material/integrations/tailwindcss/tailwindcss-v4.md

```diff
@@ -0,0 +1,132 @@
+
+To integrate Tailwind CSS v4 with Material UI in a Next.js Pages Router project, start by configuring Material UI with Next.js in the [Pages Router integration guide](/material-ui/integrations/nextjs/#pages-router).
+Then follow these steps:
+
+1. Enable the [CSS layer feature](/material-ui/integrations/nextjs/#configuration-2) in a custom `_document`:
+
+```tsx title="pages/_document.tsx"
+import {
+  createCache,
+  documentGetInitialProps,
+} from '@mui/material-nextjs/v15-pagesRouter';
+
+// ...
+
+MyDocument.getInitialProps = async (ctx: DocumentContext) => {
+  const finalProps = await documentGetInitialProps(ctx, {
+    emotionCache: createCache({ enableCssLayer: true }),
+  });
+  return finalProps;
+};
+```
+
+2. Configure the layer order with the `GlobalStyles` component—it must be the first child of the `AppCacheProvider`:
+
+```tsx title="pages/_app.tsx"
+import { AppCacheProvider } from '@mui/material-nextjs/v15-pagesRouter';
+import GlobalStyles from '@mui/material/GlobalStyles';
+
+export default function MyApp(props: AppProps) {
+  const { Component, pageProps } = props;
+  return (
+    <AppCacheProvider {...props}>
+      <GlobalStyles styles="@layer theme, base, mui, components, utilities;" />
+      {/* Your app */}
+    </AppCacheProvider>
+  );
+}
+```
+
+### Vite.js or any other SPA
+
+To integrate Tailwind CSS v4 with Material UI in a Vite-based app, make the following changes in `src/main.tsx`:
+
+1. Pass the `enableCssLayer` prop to the `StyledEngineProvider` component.
+2. Configure the layer order with the `GlobalStyles` component.
+
+```tsx title="main.tsx"
+import { StyledEngineProvider } from '@mui/material/styles';
+import GlobalStyles from '@mui/material/GlobalStyles';
+
+ReactDOM.createRoot(document.getElementById('root')!).render(
+  <React.StrictMode>
+    <StyledEngineProvider enableCssLayer>
+      <GlobalStyles styles="@layer theme, base, mui, components, utilities;" />
+      {/* Your app */}
+    </StyledEngineProvider>
+  </React.StrictMode>,
+);
+```
+
+## Tailwind CSS IntelliSense for VS Code
+
+Follow the [Tailwind CSS IntelliSense installation guide](https://tailwindcss.com/docs/editor-setup#intellisense-for-vs-code) and then add the following configuration to your [VS Code `settings.json`](https://code.visualstudio.com/docs/editor/settings#_settings-json-file) file:
+
+```json
+{
+  // ...config
+  "tailwindCSS.experimental.classRegex": [["className\\s*:\\s*['\"]([^'\"]*)['\"]"]]
+}
+```
+
+This enables Tailwind CSS IntelliSense when customizing slots in Material UI components as shown in the image below.
+
+![A preview of Tailwind CSS Intellisense](/static/material-ui/tailwind-intellisense.jpg)
+
+## Usage
+
+- Use the `className` prop to apply Tailwind CSS classes to the root element of the component.
+- Use `slotProps.{slotName}.className` to apply Tailwind CSS classes to a component's [interior slots](/material-ui/customization/overriding-component-structure/#interior-slots).
```

Comment:
We introduce the concept of interior slots in the IntelliSense section before describing why it matters in this Usage section. I think we should probably move this above the IntelliSense section.

Reply:
Got you. I wrote the docs based on my user experience of the setup.
I'd say I rather have everything setup before start using Tailwind with Material UI.

I afraid that if the "Usage" comes before, users might not realize about the extension setup when they don't see autocomplete of Tailwind classes in the slots.